### ISSUE TITLE
The gossiped view update backlog is only read on shard 0

### ISSUE THREAD

**Issue by wmitros** [scylladb/scylladb#19232]
The view update backlogs, stored for all other nodes have a copy for each shard (because they're a member of the sharded `storage_proxy`):
https://github.com/scylladb/scylladb/blob/dfd996e7c1ed6346b1edbc023addddb6b49fa107/service/storage_proxy.hh#L294

Currently, they're only used for the mv flow control, but with https://github.com/scylladb/scylladb/pull/18334 they will also be used for admission control. With admission control, if a node gets overwhelmed, we will stop sending writes to it, until it's view update backlog drops. However, we can only notice this drop with a replica write response (which we may be no longer sending) or using the gossip mechanism, where we propagate the backlog every 1s to all nodes.

Currently, the gossiped backlog is only read on shard 0, and it's not shared with other shards. As a result, with admission control, on shards other than 0, we may reach a point where the backlog of some node is too big to allow sending updates and we never lower it, causing some writes to be impossible to complete.

This can be fixed in a few ways:
1. Change the admission control, so that when we decide that a node is overloaded, we still send an empty write to it to get its backlog in the response.
2. Share the backlog received on shard 0 from gossip with other shards
3. Have all shards use the same backlogs for other nodes, perhaps by aggregating values received on all shards every 10ms (like what we're doing when sending the backlog of a node)
4. Gossip the view update backlog on all shards
5. Expire backlogs received from other nodes after some time

In approaches 1. and 5. the gossip will be mostly useless, as it would only ever affect the shard 0.
In the remaining approaches, we may still encounter the issue of a blocked (overloaded) replica, but only until the next gossip round is performed (1s).

**Comment by wmitros** [comment]
cc @piodul @nyh

**Comment by piodul** [comment]
The original doc for view backpressure (https://docs.google.com/document/d/1J6GeLBvN8_c3SbLVp8YsOXHcLc9nOLlRY7pC6MH3JWo/edit?usp=sharing) suggests that coordinator was not shard aware at some point... The doc was written before I started being a ScyllaDB developer, so I don't know what it exactly means, but it sounds like propagating this value to other shards was simply unneeded back then. There is even a suggestion that, when the coordinator becomes shard aware, we should keep a per-shard backlog, actually (now we are propagating a maximum of all shards, which is more conservative but also works).

The fact that we take the backlog we got from gossip and update local structures only on shard 0 sounds like an oversight for me. I think we should propagate it to all shards.
